test/ptc/integration/integration_test.clj

```diff
@@ -17,28 +20,55 @@
               (testing "upload a file to the bucket"
                 (let [object (str prefix "/deps.edn")]
                   (try
-                    (gcs/upload-file "deps.edn" bucket object)
+                    (gcs-tools/upload-file "deps.edn" bucket object)
                     (gcs/list-objects bucket object)
-                    (finally (gcs/delete-object bucket object)))))
-              ;; to break out from the loop
-              false)
+                    (finally (gcs-tools/delete-object bucket object)))))
+              false)                    ; to break out from the loop
             (flow [connection queue]
               (start/produce connection queue "text" properties)
               (start/listen-and-consume-from-queue task connection queue))]
       (testing "Message is not nil and can be properly read"
-        (if-let [msg (jms-tools/with-test-queue-connection flow)]
-          (is (= @jms-tools/good-jms-message (select-keys msg [::jms/Properties])))
+        (if-let [msg (jms-tools/call-with-test-connection "queue" flow)]
+          (is (= @jms-tools/good-jms-message
+                 (select-keys msg [::jms/Properties])))
           (is false))))))
 
 (deftest peeking
   (let [properties (::jms/Properties (jms/encode @jms-tools/good-jms-message))]
     (letfn [(task [message _] (is message) false)]
-      (jms-tools/with-test-queue-connection
-        (fn [connection queue]
-          (testing "Message given to task isn't nil"
-            (start/produce connection queue "text" properties)
-            (start/listen-and-consume-from-queue task connection queue))
-          (testing "The message was only peeked and can still be consumed"
-            (let [msg (jms/ednify (start/consume connection queue))]
-              (testing "Message is not nil and can be properly read"
-                (is (= @jms-tools/good-jms-message (select-keys msg [::jms/Properties])))))))))))
+      (jms-tools/call-with-test-connection
+       "queue"
+       (fn [connection queue]
+         (testing "Message given to task isn't nil"
+           (start/produce connection queue "text" properties)
+           (start/listen-and-consume-from-queue task connection queue))
+         (testing "The message was only peeked and can still be consumed"
+           (let [msg (jms/ednify (start/consume connection queue))]
+             (testing "Message is not nil and can be properly read"
+               (is (= @jms-tools/good-jms-message
+                      (select-keys msg [::jms/Properties])))))))))))
+
+(deftest forwarding
+  (testing "forward a bad message to the dead-letter queue"
```

Comment:
Am I right that this integration test has no assertions?  What would the behavior be if the DLQ forwarding didn't work as expected?

Reply:
Ah, I see now: the assertions are in `ok?`.

Reply:
`ok?` asserts.

Reply:
Yeah. Your question prompted me to lift the `is` out to make it more obvious.
I think that is generally good style in tests.